### ISSUE TITLE
Crash on startup

### ISSUE THREAD

**Issue by OgelGames** [minetest-mods/moreblocks#170]
After https://github.com/minetest-mods/moreblocks/commit/ddf8b39f5a7e47ab8aa2c09f26108d6edc573fad, mod crashes on startup:

```
2020-12-31 05:43:36: ERROR[Main]: ModError: Failed to load and run script from /var/lib/minetest/.minetest/worlds/world/worldmods/moreblocks/init.lua:
2020-12-31 05:43:36: ERROR[Main]: .../worlds/world/worldmods/moreblocks/redefinitions.lua:23: attempt to index field '?' (a nil value)
2020-12-31 05:43:36: ERROR[Main]: stack traceback:
2020-12-31 05:43:36: ERROR[Main]: 	.../worlds/world/worldmods/moreblocks/redefinitions.lua:23: in function 'reconstruct_internal_craft'
2020-12-31 05:43:36: ERROR[Main]: 	.../worlds/world/worldmods/moreblocks/redefinitions.lua:30: in function 'change_recipe_amount'
2020-12-31 05:43:36: ERROR[Main]: 	.../worlds/world/worldmods/moreblocks/redefinitions.lua:51: in function 'increase_craft_production'
2020-12-31 05:43:36: ERROR[Main]: 	.../worlds/world/worldmods/moreblocks/redefinitions.lua:68: in function 'increase_craft_production_table'
2020-12-31 05:43:36: ERROR[Main]: 	.../worlds/world/worldmods/moreblocks/redefinitions.lua:76: in main chunk
2020-12-31 05:43:36: ERROR[Main]: 	[C]: in function 'dofile'
2020-12-31 05:43:36: ERROR[Main]: 	....minetest/worlds/world/worldmods/moreblocks/init.lua:23: in main chunk
```

**Comment by OgelGames** [comment]
@Oblomov

**Comment by Oblomov** [comment]
Hm this issue seems to be related to the interaction with a different mod. Can you please identify which mod is causing the issue?

**Comment by OgelGames** [comment]
I don't know, the above is from the CI for our mod collection, specifically this one: https://github.com/pandorabox-io/pandorabox-mods/pull/1286/checks?check_run_id=1629154050

**Comment by Oblomov** [comment]
Jeepers that's a lot of mods. However, I cannot reproduce with my setup, so I need someone to provide me with enough information do a repro, possibly without testing _all_ of those mods. You probably want to look at mods that redefine/enhance signs or rails.

**Comment by OgelGames** [comment]
I modified the code to print out what recipes it was changing, and tested it locally with all the mods, and I found that it was this recipe that was causing it to crash: https://github.com/minetest-mirrors/ehlphabet/blob/277f31d6053a9ff6b75a76edf93e8f9cc7a492be/init.lua#L285-L289

Interestingly, this is how the recipe was dumped:
```
{
        width = 0,
        type = "normal",
        items = {
                "ehlphabet:block"
        },
        output = "default:paper",
        method = "normal"
}
```

Maybe the width should be checked, because that was the only recipe changed that had `width = 0` ?

**Comment by Oblomov** [comment]
OK this should be fixed in #171. Of course the weird thing in this case is that crafting and de-crafting these blocks will basically provide infinite paper. We should probably have a mechanism to prevent some recipes from being pumped up. I'll open a separate issue about this.
